google/cloud/storage/internal/curl_resumable_upload_session_test.cc

```diff
@@ -78,16 +78,16 @@ TEST(CurlResumableUploadSessionTest, Simple) {
                   "none-match-etag");
         EXPECT_EQ(test_url, request.upload_session_url());
         EXPECT_THAT(request.payload(), MatchesPayload(payload));
-        EXPECT_EQ(0, request.source_size());
-        EXPECT_EQ(0, request.range_begin());
+        EXPECT_FALSE(request.upload_size().has_value());
+        EXPECT_EQ(0, request.offset());
         return make_status_or(ResumableUploadResponse{
             "", ResumableUploadResponse::kInProgress, size, {}, {}});
       })
       .WillOnce([&](UploadChunkRequest const& request) {
         EXPECT_EQ(test_url, request.upload_session_url());
         EXPECT_THAT(request.payload(), MatchesPayload(payload));
-        EXPECT_EQ(2 * size, request.source_size());
-        EXPECT_EQ(size, request.range_begin());
+        EXPECT_EQ(2 * size, request.upload_size().value_or(0));
```

Comment:
Nit: Below we also expect `request.upload_size().has_value()` before the `value_or()`.

Reply:
Added here, though it is implicit (here, but not below) as it compares to a different value than the `0` passed to `value_or()`.